### ISSUE TITLE
params.circularfilter ? '' : '-f true -x false'

### ISSUE THREAD

**Issue by zhongxing123** [nf-core/eager#638]
Hello，I'm a little confused about circularmapper.  I find script `def filter = params.circularfilter ? '' : '-f true -x false'` in main.nf. Does this mean that `-f true-x false` is performed when `circularfilter == false`? Please give me some guidance. This has been bothering me for a long time. Thank you very much

**Comment by apeltzer** [comment]
Hi @zhongxing123 ! Yes, this means when the parameter `--circularfilter` is set to `false`, that the pipeline will be run with the `-f true -x false` option(s).

**Comment by zhongxing123** [comment]
Hi Dr. Alexander Peltzer ! Dose the -f true -x false option(s) mean turn on the filtering? But the parameter --circularfilter setted to false mean turn off the filtering. I don’t know if I got it right, I hope you can 
Help me solve it again. Thank you very much.
Xing Zhang
College of Animal Science and Technology, China Agricultural University No.2 Yuanmingyuan West Road, Haidian District, Beijing 100193, P. R. China Mobile: (+86)18200452413  WeChat: daxian7144

发件人: Alexander Peltzer
发送时间: 2020年12月19日 23:09.
收件人: nf-core/eager
抄送: zhongxing123; Mention
主题: Re: [nf-core/eager] params.circularfilter ? '' : '-f true -x false'(#638)

Hi @zhongxing123 ! Yes, this means when the parameter --circularfilter is set to false, that the pipeline will be run with the -f true -x false option(s).

**Comment by jfy133** [comment]
@apeltzer actually isn't this a bug?

``` 
def filter = params.circularfilter ? '' : '-f true -x false'
```

If `--circularfilter` is TRUE, you would want filtering to turn, correct? Therefore if TRUE you _would_ want  to supply `-f` (to remove the reads that don't match the circular correct? At least that's what I understand from `realignsamfile`?

```
 -f,--filterCircularReads <FILTER>                     filter the reads
                                                       that don't map to a
                                                       circular identifier
                                                       (true/false),
                                                       default false
```

**Comment by jfy133** [comment]
Hi @zhongxing123 -

Thank you for looking so closely at the code. Indeed your confusion may have stemmed from the fact this was indeed a bug.

in the fix #639  we have corrected this. So in the next release:

`--circularfilter` will turn ON filtering of circularmapper generated BAM file. This will therefore produce BAM files with reads that _only_ mapped to the elongated reference genome, and also now with the non-elongated references to be excluded from the BAM header.

**Comment by zhongxing123** [comment]
Thank you for being so careful to help me deal with this problem, which is very useful to me. Thanks again for your help.


Xing Zhang
College of Animal Science and Technology, China Agricultural University No.2 Yuanmingyuan West Road, Haidian District, Beijing 100193, P. R. China Mobile: (+86)18200452413  WeChat: daxian7144

发件人: James A. Fellows Yates
发送时间: 2020年12月21日 17:42
收件人: nf-core/eager
抄送: zhongxing123; Mention
主题: Re: [nf-core/eager] params.circularfilter ? '' : '-f true -x false'(#638)

@apeltzer actually isn't this a bug?
def filter = params.circularfilter ? '' : '-f true -x false'
If --circularfilter is TRUE, you would want filtering to turn, correct? Therefore if TRUE you would want to supply -f (to remove the reads that don't match the circular correct? At least that's what I understand from realignsamfile?
 -f,--filterCircularReads <FILTER>                     filter the reads
                                                       that don't map to a
                                                       circular identifier
                                                       (true/false),
                                                       default false
